internal/commands/app/create_inputs.go

```diff
@@ -88,29 +85,28 @@ func (i *createInputs) resolveDirectory(wd string) (string, error) {
 		return "", appErr
 	}
 	if appOK {
-		return "", fmt.Errorf("%s is inside or is a Realm app directory", dir)
+		return "", fmt.Errorf("%s is inside or is a Realm app directory", fullPath)
 	}
 	return fullPath, nil
 }
 
-// TODO(REALMC-8135): Implement data-source flag for app create command
-// func (i *createInputs) resolveDataSource(client realm.Client, groupID, appID string) (string, error) {
-// 	clusters, err := client.ListClusters(groupID, appID)
-// 	if err != nil {
-// 		return "", err
-// 	}
-// 	var dsCluster string
-// 	for _, cluster := range clusters {
-// 		if (i.DataSource == "" && cluster.State == "IDLE") || i.DataSource == cluster.Name {
-// 			dsCluster = cluster.Name
-// 			break
-// 		}
-// 	}
-// 	if dsCluster == "" {
-// 		if i.DataSource != "" {
-// 			return "", fmt.Errorf("Unable to find the %s cluster", i.DataSource)
-// 		}
-// 		return "", fmt.Errorf("Unable to find any available cluster for Group ID %s", groupID)
-// 	}
-// 	return dsCluster, nil
-// }
+func (i *createInputs) resolveDataSource(client realm.Client, groupID, appID string) (string, error) {
+	clusters, err := client.ListClusters(groupID, appID)
+	if err != nil {
+		return "", err
+	}
+	var dsCluster string
+	for _, cluster := range clusters {
+		if (i.DataSource == "" && cluster.State == "IDLE") || i.DataSource == cluster.Name {
+			dsCluster = cluster.Name
+			break
+		}
+	}
+	if dsCluster == "" {
+		if i.DataSource != "" {
+			return "", fmt.Errorf("unable to find the %s cluster", i.DataSource)
+		}
+		return "", fmt.Errorf("unable to find any available cluster for Project %s", groupID)
```

Comment:
thoughts on simplifying this and just returning `errors.New("failed to find Atlas cluster")` here?  in the context of running this command, the user probably already know the name they specified

Reply:
I agree, though I do like the difference in error between failing to find one they specified vs any available cluster

Reply:
to close the loop here as well, this scenario does not exist with the current implementation.

should we ever support these bool/string hybrid-type flags, we can revisit this